modules/loader/src/main/java/com/opengamma/strata/loader/csv/FxSingleBarrierOptionTradeCsvPlugin.java

```diff
@@ -0,0 +1,144 @@
+import com.opengamma.strata.loader.LoaderUtils;
+import com.opengamma.strata.product.Trade;
+import com.opengamma.strata.product.TradeInfo;
+import com.opengamma.strata.product.fxopt.FxSingleBarrierOption;
+import com.opengamma.strata.product.fxopt.FxSingleBarrierOptionTrade;
+import com.opengamma.strata.product.fxopt.FxVanillaOptionTrade;
+import com.opengamma.strata.product.option.BarrierType;
+import com.opengamma.strata.product.option.KnockType;
+import com.opengamma.strata.product.option.SimpleConstantContinuousBarrier;
+
+/**
+ * Handles the CSV files format for FX Single Barrier Option trades.
+ */
+public class FxSingleBarrierOptionTradeCsvPlugin implements TradeCsvParserPlugin, TradeCsvWriterPlugin<FxSingleBarrierOptionTrade> {
+
+  /** The singleton instance of the plugin. */
+  public static final FxSingleBarrierOptionTradeCsvPlugin INSTANCE = new FxSingleBarrierOptionTradeCsvPlugin();
+
+  /** The CSV headers. */
+  public static final Set<String> HEADERS = ImmutableSet.<String>builder()
+      .addAll(FxVanillaOptionTradeCsvPlugin.INSTANCE.headers(ImmutableList.of()))
+      .add(BARRIER_TYPE_FIELD)
+      .add(KNOCK_TYPE_FIELD)
+      .add(BARRIER_LEVEL_FIELD)
+      .add(REBATE_AMOUNT_FIELD)
+      .add(REBATE_CURRENCY_FIELD)
+      .add(REBATE_DIRECTION_FIELD)
+      .build();
+
+  /** Restricted constructor */
+  private FxSingleBarrierOptionTradeCsvPlugin(){
+  }
+
+  @Override
+  public Set<String> tradeTypeNames() {
+    return ImmutableSet.of("FXSINGLEBARRIEROPTION", "FX SINGLE BARRIER OPTION");
+  }
+
+  @Override
+  public Optional<Trade> parseTrade(
+      Class<?> requiredJavaType,
+      CsvRow baseRow,
+      List<CsvRow> additionalRows,
+      TradeInfo info,
+      TradeCsvInfoResolver resolver) {
+
+    if (requiredJavaType.isAssignableFrom(FxSingleBarrierOptionTrade.class)) {
+      return Optional.of(parse(baseRow, info, resolver));
+    }
+    return Optional.empty();
+  }
+
+  private FxSingleBarrierOptionTrade parse(CsvRow row, TradeInfo info, TradeCsvInfoResolver resolver) {
+    FxVanillaOptionTrade vanillaOptionTrade = resolver.parseFxVanillaOptionTrade(row, info);
+
+    BarrierType barrierType = row.getValue(BARRIER_TYPE_FIELD, LoaderUtils::parseBarrierType);
+    KnockType knockType = row.getValue(KNOCK_TYPE_FIELD, LoaderUtils::parseKnockType);
+    double barrierLevel = row.getValue(BARRIER_LEVEL_FIELD, LoaderUtils::parseDouble);
+    AdjustablePayment premium = CsvLoaderUtils.parsePremiumFromDefaultFields(row);
+
+    FxSingleBarrierOption.Builder productBuilder = FxSingleBarrierOption.builder()
+        .underlyingOption(vanillaOptionTrade.getProduct())
+        .barrier(SimpleConstantContinuousBarrier.of(barrierType, knockType, barrierLevel));
+
+    if (CsvLoaderUtils.hasValidCurrencyAmount(
+        row,
+        REBATE_CURRENCY_FIELD,
+        REBATE_AMOUNT_FIELD,
+        REBATE_DIRECTION_FIELD)) {
+      productBuilder.rebate(
+          CsvLoaderUtils.parseCurrencyAmountWithDirection(row,
```

Comment:
This is a case where adding a method in `CsvLoaderUtils` called `tryParseCurrencyAmountWithDirection` returning an `Optional` would make sense.